src/crucible/aws/cloudwatch/anomaly_detector.clj

```diff
@@ -0,0 +1,45 @@
+(ns crucible.aws.cloudwatch.anomaly-detector
+  (:require [clojure.spec.alpha :as s]
+            [crucible.aws.cloudwatch :as cloudwatch]
+            [crucible.encoding.keys :refer [->key]]
+            [crucible.resources :refer [defresource spec-or-ref]]))
+
+(s/def ::value (spec-or-ref string?))
+
+(s/def ::name (spec-or-ref string?))
+
+(s/def ::dimension (s/keys :req [::name ::value]))
+
+(s/def ::dimensions (s/coll-of ::dimension :kind vector?))
+
+(s/def ::stat ::cloudwatch/statistic)
+
+(def date-time-regex
+  #"(19|20)[0-9][0-9]-(0[0-9]|1[0-2])-(0[1-9]|([12][0-9]|3[01]))T([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9]")
+
+(s/def ::date-time (spec-or-ref (s/and string? #(re-matches date-time-regex %))))
+
+(s/def ::end-time ::date-time)
+
+(s/def ::start-time ::date-time)
+
+(s/def ::range (s/keys :req [::start-time ::end-time]))
+
+(s/def ::ranges (s/coll-of ::range))
+
+(s/def ::name-space (spec-or-ref string?))
```

Comment:
`namespace` is a happy compound word in clojure. Then you wouldn't have to `->key` it.

Reply:
Feedback addressed in https://github.com/brabster/crucible/pull/188/commits/1c698299a9b9a2768de0b59360f37a07cc0186df